packages/gatsby-theme-docs/utils/bootstrap-theme-addons.js

```diff
@@ -17,9 +17,10 @@ const gitignorePath = path.join(addOnsFolderPath, '.gitignore');
 
 function bootstrapThemeAddOnsForMarkdownComponents(themeAddOnNames) {
   let addOnComponentsExport = `/* eslint-disable prettier/prettier */
+/* eslint import/no-anonymous-default-export: [2, {"allowObject": true}] */
 /* THIS IS AN AUTOGENERATED FILE, DO NOT EDIT DIRECTLY */`;
   themeAddOnNames.forEach((plugin, index) => {
-    addOnComponentsExport += `\nimport * as components${index} from '${plugin}';`;
+    addOnComponentsExport += `\nimport * as components${index} from '${plugin}/shortcodes';`;
```

Comment:
important change

Reply:
I'm starting to wonder if we really need this change. The point of the add-ons was that they export components that are automatically injected into the provider.

From what I understand here, we're trying to do the same thing except for the "api-docs" add-on, which is not an add-on anymore since the change here is to import the api components explicitly.

This all export `index.js` vs `shortcodes.js` is also a bit confusing and to me it feels like a symptom to the underlying problem.

Therefore, given that we already agreed that we will have explicit imports in the MDX files, how about we do this for all add-ons and not just the api-docs?

Advantages:
* Same behavior across MDX add-ons.
* No bloated MDX provider.
* Each website will explicitly define a `shortcodes.js` with components that should be imported from the different theme packages.
* Each theme package has normal exports via `index.js`.

This obviously means that there won't be any "add-ons" with automatic imports/overrides but "just normal packages".

Would that be an option to have things more consistent? @nkuehn

Reply:
You're right in that our understanding of an add-on is including the automatic injection of components into the provider.  But I don't see it as an exclusive criterion.  To me, an add-on theme is a gatsby theme that only works in conjunction with the base theme and is assuming all the infrastructure and UI framing of the base theme.    It could for example not add new MDX components but rather override some other part of the base theme through component shadowing and would still be an add on. 

I understand your concern about consistency but I do not think the focus on consistency is improving the product in this case.  
The majority of documentation websites is not using the API docs add on and the long tail of users does not interact with it, too.   I don't see a point in making the system harder to use in the majority case and debug "just" for consistency in the codebase.